### PR TITLE
Create test contract AuthorStake.t.sol

### PR DESCRIPTION
**Pull request by krisoshea-eth**
This PR adds the AuthorStake.t.sol test file to the repo. This test file contains all the relevant foundry tests for the AuthorStake.sol contract.

### CODE REVIEW

File: test/AuthorStake.t.sol [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+
+        // Attempt to withdraw more than the staked amount should fail
+        vm.expectRevert(bytes(""));
+
+        stakingContract.withdrawStake(withdrawAmount);
+        vm.stopPrank();
+    }
+
+    function testFailWithdrawalWithoutStaking() public {
```
Comment (RojhatToptamus) [inline test/AuthorStake.t.sol:103]: same here

File: test/AuthorStake.t.sol [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        assertEq(
+            stakedBalance,
+            stakeAmount,
+            "Staked amount does not match expected balance"
+        );
+    }
+
+    function testFuzzWithdrawalAmounts(uint256 _withdrawAmount) public {
+        // Ensure there's a non-zero amount to work with, adjusted to avoid excessively large transactions
+        uint256 withdrawAmount = (_withdrawAmount % 1 ether) + 1; // Ensure at least 1 wei is withdrawn and cap at 1 ether for practicality
```
Comment (RojhatToptamus) [inline test/AuthorStake.t.sol:262]: This isn't necessary, and we can limit the amount using the **vm.assume** keyword.

File: test/AuthorStake.t.sol [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+        uint256 gasBefore = gasleft();
+        stakingContract.stakeEtherFrom{value: stakeAmount}(staker);
+        uint256 gasAfter = gasleft();
+        uint256 gasUsed = gasBefore - gasAfter;
+        vm.stopPrank();
+    
+        console2.log("Gas used for staking 1 ether:", gasUsed);
+    }
+    
+    function testWithdrawalExceedsStakedAmountWithRevertReason() public {
```
Comment (RojhatToptamus) [inline test/AuthorStake.t.sol:352]: This test case will no longer be needed, after fixing the issues in the **testFailWithdrawalExceedsStakedAmount**.

**Comment by KohliSuraj** [comment]
@RojhatToptamus Good shout figuring out the issue 👍 Can we change the test function names to follow naming conventions best practices - https://book.getfoundry.sh/tutorials/best-practices#tests

**Review by KohliSuraj — APPROVED** [review]

**Pull request merged**